shared/dataflowstack/codeql/dataflowstack/DataFlowStack.qll

```diff
@@ -0,0 +1,372 @@
+
+private import codeql.dataflow.DataFlow as DF
+
+module DataFlowStackMake<DF::InputSig Lang>{
+
+    import DF::DataFlowMake<Lang> as DataFlow
+
+    module BiStackAnalysis<DataFlow::GlobalFlowSig FlowA, DataFlow::GlobalFlowSig FlowB>{
+
+        module FlowStackA = FlowStack<FlowA>;
+        module FlowStackB = FlowStack<FlowB>;
+
+        /**
+         * Holds if either the Stack associated with `sourceNodeA` is a subset of the stack associated with `sourceNodeB`
+         * or vice-versa.
+         */
+        predicate eitherStackSubset(FlowA::PathNode sourceNodeA, FlowA::PathNode sinkNodeA, FlowB::PathNode sourceNodeB, FlowB::PathNode sinkNodeB){
+            sourceNodeA.isSource() and
+            sourceNodeB.isSource() and
+            not exists(sinkNodeA.getASuccessor()) and
+            not exists(sinkNodeB.getASuccessor()) and
```

Comment:
FlowA::isSource / FlowA::isSink

Reply:
No `isSink` defined in common PathNode signature.  I want to make very limited changes to the common signatures for now to speed up merging, so I have defined the refactored predicate under `FlowStack` for now with a note to refactor later.